crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -217,6 +373,31 @@ impl<P: FsProvider + Send + Sync + 'static> Plugin for ResctrlPlugin<P> {
         req: UpdateContainerRequest,
     ) -> ttrpc::Result<UpdateContainerResponse> {
         debug!("resctrl-plugin: update_container: {}", req.container.id);
+        if let Some(pod) = req.pod.as_ref() {
+            let pod_uid = pod.uid.clone();
+            if let Ok(group_path) = self.resctrl.create_group(&pod_uid) {
+                let container_opt: Option<nri::api::Container> = req.container.as_ref().cloned();
+                let passes = self.cfg.max_reconcile_passes;
+                let result = self.resctrl.reconcile_group(
+                    &group_path,
+                    || {
+                        let v: Vec<nri::api::Container> =
+                            container_opt.as_ref().map(|c| vec![c.clone()]).unwrap_or_default();
+                        Ok(self.enumerate_pod_pids(&v))
+                    },
+                    passes,
+                );
+                let new_state = match result {
+                    Ok(ar) => if ar.missing == 0 { AssignmentState::Success } else { AssignmentState::Partial },
+                    Err(e) => { warn!("resctrl-plugin: reconcile (update_container) failed: {}", e); AssignmentState::Partial },
+                };
+                if let Some(ev) = self.should_emit_and_update(&pod_uid, new_state, Some(group_path)) {
+                    self.emit_event(ev);
+                }
+            } else if let Some(ev) = self.should_emit_and_update(&pod_uid, AssignmentState::Failure, None) {
+                self.emit_event(ev);
+            }
+        }
         Ok(UpdateContainerResponse::default())
     }
 
```

Comment:
I don't see that we handle cleanup on pod removal. those events are handled as part of the state change handle, see for example https://github.com/containerd/nri/blob/10ffc61c61e331e6f20ec80a509ef939f83f2b1e/pkg/stub/stub.go#L772 (that is go not rust, but is informative)